presto-hive/src/main/java/io/prestosql/plugin/hive/HivePageSource.java

```diff
@@ -631,7 +626,14 @@ public IntArrayList computeEligibleRowIds(Page page)
                     ids.add(position);
                 }
             }
-            return ids;
+            int keptRowCount = ids.size();
+            if (keptRowCount == 0) {
+                return null; // Empty page after filtering
+            }
+            if (keptRowCount == page.getPositionCount()) {
+                return page; // Unchanged after filtering
```

Comment:
i think the comments are redundant, but up to you

Reply:
Removed.